src/abstract/CSBondCurve.sol

```diff
@@ -152,67 +148,61 @@ abstract contract CSBondCurve is ICSBondCurve, Initializable {
-            if (curvePoints[i] <= curvePoints[i - 1])
-                revert InvalidBondCurveValues();
-        }
+
+        _checkBondCurve(curvePoints);
+
         uint256 curveTrend = curvePoints[curvePoints.length - 1] -
             // if the curve length is 1, then 0 is used as the previous value to calculate the trend
             (curvePoints.length > 1 ? curvePoints[curvePoints.length - 2] : 0);
         $.bondCurves.push(
-            BondCurve({
-                id: $.bondCurves.length + 1, // to avoid zero id in arrays
-                points: curvePoints,
-                trend: curveTrend
-            })
+            BondCurve({ points: curvePoints, trend: curveTrend })
         );
         emit BondCurveAdded(curvePoints);
-        return $.bondCurves.length;
+        return $.bondCurves.length - 1;
     }
 
-    /// @dev Set default bond curve for the module
-    ///      It will be used for the Node Operators without a custom curve set
-    function _setDefaultBondCurve(uint256 curveId) internal {
+    /// @dev Update existing bond curve
+    function _updateBondCurve(
+        uint256 curveId,
+        uint256[] memory curvePoints
+    ) internal {
         CSBondCurveStorage storage $ = _getCSBondCurveStorage();
-        if (
-            curveId == 0 ||
-            curveId > $.bondCurves.length ||
-            curveId == $.defaultBondCurveId
-        ) revert InvalidBondCurveId();
-        $.defaultBondCurveId = curveId;
-        emit DefaultBondCurveChanged(curveId);
+        if (curveId > $.bondCurves.length - 1) revert InvalidBondCurveId();
+
+        _checkBondCurve(curvePoints);
+
+        uint256 curveTrend = curvePoints[curvePoints.length - 1] -
+            // if the curve length is 1, then 0 is used as the previous value to calculate the trend
+            (curvePoints.length > 1 ? curvePoints[curvePoints.length - 2] : 0);
+        $.bondCurves[curveId] = BondCurve({
+            points: curvePoints,
+            trend: curveTrend
+        });
+        emit BondCurveUpdated(curveId, curvePoints);
     }
 
     /// @dev Sets bond curve for the given Node Operator
     ///      It will be used for the Node Operator instead of the default curve
     function _setBondCurve(uint256 nodeOperatorId, uint256 curveId) internal {
         CSBondCurveStorage storage $ = _getCSBondCurveStorage();
-        if (curveId == 0 || curveId > $.bondCurves.length)
-            revert InvalidBondCurveId();
+        if (curveId > $.bondCurves.length - 1) revert InvalidBondCurveId();
         $.operatorBondCurveId[nodeOperatorId] = curveId;
         emit BondCurveChanged(nodeOperatorId, curveId);
     }
 
     /// @dev Reset bond curve for the given Node Operator to default (for example, because of breaking the rules by Node Operator)
     function _resetBondCurve(uint256 nodeOperatorId) internal {
         CSBondCurveStorage storage $ = _getCSBondCurveStorage();
-        delete $.operatorBondCurveId[nodeOperatorId];
```

Comment:
Since we assume 0 as a default value it can be a constant value in the contract.

Reply:
Done